src/dune_rules/melange/melange_rules.ml

```diff
@@ -484,6 +484,13 @@ let setup_js_rules_libraries =
   in
   fun ~dir ~scope ~target_dir ~sctx ~requires_link ~mode (mel : Melange_stanzas.Emit.t) ->
     let build_js = build_js ~sctx ~mode ~module_systems:mel.module_systems in
+    let vlib_implementations =
```

Comment:
Maybe instead of getting the `vlib_implementations` here and below define some inline callback that uses them, I would create a `with_vlib_implementations` function here that takes the `deps` arg, so that it can be used directly below, e.g.:

```ml
          |> Resolve.Memo.map ~f:with_vlib_implementations
```

Reply:
but then you'd be creating it on every iteration of the loop, no?

Reply:
No, I mean something like this, in line 486-487 (outside the loop):

```ml
  fun ~dir ~scope ~target_dir ~sctx ~requires_link ~mode (mel : Melange_stanzas.Emit.t) ->
    let build_js = build_js ~sctx ~mode ~module_systems:mel.module_systems in
    let with_vlib_implementations deps = ...all the new stuff..
```

Then in the loop just this

```ml
          |> Resolve.Memo.map ~f:with_vlib_implementations
```

Reply:
It's pretty minor in any case.

Reply:
check what I've pushed. is that what you had in mind?

Reply:
Exactly! 👍